src/main/java/seedu/address/logic/commands/UndoableCommand.java

```diff
@@ -5,14 +5,18 @@
 import static seedu.address.model.Model.PREDICATE_SHOW_ALL_BOOKS;
 
 import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ActiveListType;
 import seedu.address.model.BookShelf;
 import seedu.address.model.ReadOnlyBookShelf;
+import seedu.address.model.book.exceptions.DuplicateBookException;
 
 /**
  * Represents a command which can be undone and redone.
  */
 public abstract class UndoableCommand extends Command {
```

Comment:
Remove changes to this file. `redo` will be removed so there's no need to try to make it work.